### PR TITLE
Declare and use the new ADAPTER constant.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-gson-support/src/test/java/com/squareup/wire/protos/alltypes/AllTypes.java [hunk 1]
```diff
@@ -18,7 +20,9 @@
 import okio.ByteString;
 
 public final class AllTypes extends Message<AllTypes> {
-  private static final long serialVersionUID = 0L;
+  public static final ProtoAdapter<AllTypes> ADAPTER = Wire.messageAdapter(AllTypes.class);
+
+  public static final long serialVersionUID = 0L;
```
Comment (JakeWharton) [inline wire-gson-support/src/test/java/com/squareup/wire/protos/alltypes/AllTypes.java]: This should (can) be `private`

File: wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 2]
```diff
@@ -345,8 +357,13 @@ public TypeSpec generateMessage(MessageType type) {
...
     builder.superclass(messageOf(javaType));
 
+    builder.addField(FieldSpec.builder(adapterOf(javaType), "ADAPTER")
+        .addModifiers(PUBLIC, STATIC, FINAL)
+        .initializer("$T.messageAdapter($T.class)", Wire.class, javaType)
+        .build());
+
     builder.addField(FieldSpec.builder(TypeName.LONG, "serialVersionUID")
-        .addModifiers(PRIVATE, STATIC, FINAL)
+        .addModifiers(PUBLIC, STATIC, FINAL)
```
Comment (JakeWharton) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: ...looks like a deliberate change though.

Comment (swankjesse) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: Nope, that's a bug.

Comment (swankjesse) [inline wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: (thought I was fixing `ADAPTER` to be public and fixed the wrong field. Derp.)

**Comment by JakeWharton** [comment]
LGTM :shipit: :shipit: :shipit: :shipit: :shipit: :shipit:

**Pull request merged**